src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,163 @@
+package duke;
+
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import java.util.Scanner;
+
+/**
+ * Class that deals with making sense of the user command
+ * @author Ashiqur Rahman A0230107Y
+ */
+public class Parser {
+
+    /**
+     * Method that echos command
+     * @param echoCommand Command to be echoed
+     */
+    public static void echo(String echoCommand) {
+        System.out.println("--------------------------------------------------");
+        System.out.println(echoCommand);
+        System.out.println("--------------------------------------------------");
+    }
+
+    /**
+     * Method that writes to file
+     * @param tasks List of task that has to written on in file
+     * @throws IOException
+     */
+    private static void writeToFile(TaskList tasks) throws IOException {
+        String filePath = "./././data/duke.txt";
+        String directoryPath = "./././data";
+        if (!Files.exists(Path.of(directoryPath))) {
+            throw new IOException("Directory Data does not exist");
+        }
+        if (!Files.exists(Path.of(filePath))) {
+            throw new IOException("File, duke.text, does not exist");
+        }
+        FileWriter fw = null;
+        try {
+            fw = new FileWriter(filePath);
+            fw.write(TaskList.printTaskList(tasks));
+            fw.close();
+        } catch (IOException e) {
+
+        }
+    }
+
+    public static void run (TaskList tasklist) throws IOException {
+        String command = "";
+        while (true) {
+            Scanner userInput = new Scanner(System.in);// Create a Scanner object
+            command = userInput.nextLine(); // Read user input
+            if ("bye".equals(command)) {
+                echo("Bye. Hope to see you again soon!");
+                break;
+            } else if ("list".equals(command)) {
+                echo(TaskList.printTaskList(tasklist));
+            } else if (command.contains("unmark")) { // to detect unmark command
+                String number = command.replaceAll("[^\\d.]", "");
+                int n = Integer.parseInt(number);
+                if (n > tasklist.size()){
+                    echo(DukeException.IndexOutofBoundsException(tasklist));
+                } else {
+                    Task unmarkedTask = tasklist.get(n-1);
+                    unmarkedTask.markAsUndone();
+                    String taskStatus = String.format("OK, I've marked this task as not done yet:\n%s", unmarkedTask);
+                    echo(taskStatus);
+                    writeToFile(tasklist);
+                }
+            } else if (command.contains("mark")){ // to detect mark command
+                String number = command.replaceAll("[^\\d.]", "");
+                int n = Integer.parseInt(number);
+                if (n > tasklist.size()){
```

Comment:
Should there be a spacing?
```suggestion
                if (n > tasklist.size()) {
```